Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -484,7 +536,7 @@ - (void)trackLimboChange:(FSTLimboDocumentChange *)limboChange {
                                                          targetID:limboTargetID
                                              listenSequenceNumber:kIrrelevantSequenceNumber
                                                           purpose:FSTQueryPurposeLimboResolution];
-    _limboKeysByTarget[limboTargetID] = key;
+    _limboResolutionsByTarget[limboTargetID] = LimboResolution{key};
```

Comment:
This default constructs the `LimboResolution` and then assigns it. Better to make this:

```c++
_limboResolutionsByTarget.emplace(limboTargetID, LimboResolution{key});
```